### DISCUSSION TITLE
How to convert some PNG to webp?

### DISCUSSION THREAD

**Discussion by WingDust** [nushell/nushell#14132] [Q&A]
I have some png to webp png images
try 
 ` ls *.png   | each {|n| magick $n.name ( $n.name | str replace 'png$' 'webp') } ` 
` ls *.png   | each {|n| magick $n.name ( $n.name | str replace 'png' 'webp') } ` 

but fail

```
# nu -v
0.82.0
```

![image](https://github.com/user-attachments/assets/f329bc24-407f-474c-b5f1-a4cd9b4633a3)

how to deal with ?

---
# 2024-10-23 new additon

![image](https://github.com/user-attachments/assets/ae20b5ff-dfc2-4327-b142-ceabb3545134)
The problem is with `str replace`

more again how to fix ?

**Comment by ghost** [comment]
sidenote: i think the imagemagick repository is a better place to discuss the usage of magick.

`magick $image "WEBP:-" | save -r ($image | str replace 'png$' 'webp')` works for me

**Comment by weirdan** [comment]
Check your quotes - you need to use `''` and not `‘‘`

**Reply by WingDust** [reply]
it is my copy problem ,in pictue, command use `''`

**Reply by weirdan** [reply]
I'm not so sure. They look like curly quotes to me, while I'd expect normal straight quotes to look the same regardless of whether it's an opening quote mark or the closing one (yours look different). Of course, it might be a fancy way your terminal renders normal quotes, but if I were you, I'd double-check that.

**Comment by lemorage** [comment]
I think the following command will work:

`ls *.png | each {|n| magick $n.name ($n.name | str replace 'png' 'webp')}`

**Reply by ghost** [reply]
It can work, but if you happen to have a file named like `whateverpngfile.png`, the substitution will be: `whateverwebpfile.webp`. With that in mind, a better approach would be to use regex actually so it doesn't match a *png* string in the middle of the file name.

**Comment by ghost** [comment]
```nu
glob *.png | each { magick $in ($in | str replace -r "png$" "webp") | ignore
```

To enable to `str replace` parse regular expressions, you need to use the `-r, --regex` flag. See: `$ help str replace` for usage and examples. It's great!
